packages/loader/container-loader/src/container.ts

```diff
@@ -584,14 +588,36 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             loader.services.subLogger,
             undefined,
             {
-                all:{
+                all: {
                     clientType, // Differentiating summarizer container from main container
                     loaderVersion: pkgVersion,
                     containerId: uuid(),
                     docId: () => this.id,
                     containerAttachState: () => this._attachState,
                     containerLoaded: () => this.loaded,
                 },
+                error: {
```

Comment:
This to me looks wrong and backwards.
First, how big is this list will become? When do we stop?
Second, that's not cheap, even with compression. That adds bandwidth, cost of storage, perf of querying, etc. It's just a matter of time when Amy Hood will comes to us and asks to reduce impact of our telemetry on bottom line (it happened with Office).

The right approach is to improve analyzes tools, not bake them into shipping product.
That gives much better flexibility (we can change analyzes / build more sophisticated analyzers as we learn more) without need to change code. As with anything, it's matter of automation.

I've spent a bunch of time initially when we put telemetry in to make it cheaper.
Based on Aria reports, I removed things like clientID because it was contributing a lot to bottom line.
Unfortunately (for some unknown to me reason) reports no longer show me impact:

https://portal.aria.ms/kusto/6a8929bcfc6d44e9b13fee392ada9cf0/tables?table=Office_Fluid_FluidRuntime_Performance

Reply:
Found at least some data:
https://portal.aria.ms/kusto/6a8929bcfc6d44e9b13fee392ada9cf0/database
Our two tables use half of all the Kusto space. Which is not that bad, but this change will likely change it substantially.

Reply:
I understand the concern, but i think the cost is worth it for errors. I don't know if this is the right set, it's probably overboard, but without this data is becomes hard, if not impossible to understand the true impact of issues, as most errors can't easily be linked back to their initial conditions, specifically message or timestamp. The ability to link back to the initial conditions in critical to knowing how frequently a error occurs, and not just how often it is hit, and also understand which version, or versions caused the error, not just hit the error. 

Looking at the above table, it looks like generic, and performance are the real space hogs, this will only affect error. If we want to save space we should focus on reducing the amount we log in those other tables, as they seem overly verbose.

Reply:
Agree with assessment, but I wish we invested into tools instead. Or at least have a plan to get there

Reply:
I really don't have a clear vision of what the tools could be, are you thinking opt-in things that run in the client context? or tools that mine our logs? or both? In terms of log mining i see this as an important step to getting a baseline for all data.  with this we should be able to reasonably have common queries for gauging impact and issue triangulation (build/hosts/...) that are currently really hard and really fragile

Reply:
I've reduced the logging, and added a bunch of comments. I hope this better captures my goal, and is more in line with the spirt of the feedback here.

Reply:
I'm thinking of "watson like" tools that do analyzes. I.e. some kind of bucketing system for critical errors (equivalent of crash in native side) with analyzer that creates buckets (i.e. groups), but also allows you to look at each "cab" (hit) with full dump of analyzer that went and collected (and analyzed) a bunch of data for this case (including session) which can be used both by human (to have better understanding of individual hit) as well as for aggregation